runtime/src/snapshot_utils.rs

```diff
@@ -1472,33 +1475,53 @@ fn untar_snapshot_in<P: AsRef<Path>>(
     archive_format: ArchiveFormat,
     parallel_divisions: usize,
 ) -> Result<UnpackedAppendVecMap> {
-    let open_file = || File::open(&snapshot_tar).unwrap();
+    let file = File::open(&snapshot_tar).unwrap();
+
+    // Map the file into immutable memory for better I/O performance
+    let mmap = unsafe { Mmap::map(&file) };
```

Comment:
if this PR makes it out of draft, consider using an existing crate that wraps the unsafe.  https://crates.io/crates/mmarinus seems like one such option

Reply:
It looks like this crate depends on libc. That makes it linux-only. Not for windows.

```
> cargo check
    Checking mmarinus v0.4.0
error[E0433]: failed to resolve: could not find `unix` in `os`
  --> C:\Users\hyi\.cargo\registry\src\github.com-1ecc6299db9ec823\mmarinus-0.4.0\src\builder.rs:11:14
   |
11 | use std::os::unix::io::{AsRawFd, RawFd};
   |              ^^^^ could not find `unix` in `os`
```

Reply:
Perhaps there's another crate, or we could consider upstreaming support for Windows.

Generally it's very undesirable to have raw `unsafe`s in the code like this

Reply:
Alternatively fall back to the original code for Windows?

Reply:
Yeah. Pushed a commit to update this. 
Unfortunately, there is a still unsafe code to convert mmap to slice. This is because the background thread in SharedBufferReader requires the reader to be static. It will requires a bit of rework on the SharedBufferReader, which we can addressed in the future?